File: php/features.php
<?php
declare(strict_types=1);

// Isports SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class IsportsFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new IsportsBaseFeature();
            case "test":
                return new IsportsTestFeature();
            default:
                return new IsportsBaseFeature();
        }
    }
}

File: py/features.py
# Isports SDK feature factory

from feature.base_feature import IsportsBaseFeature
from feature.test_feature import IsportsTestFeature


def _make_feature(name):
    features = {
        "base": lambda: IsportsBaseFeature(),
        "test": lambda: IsportsTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
